### DISCUSSION TITLE
Estimate memory usage in advance

### DISCUSSION THREAD

**Discussion by hsheldon** [idaholab/moose#20022] [Q&A General] [answered]
Is there a rule-of-thumb for estimating the likely memory usage of a MOOSE problem, based on the DOF? I understand there are post processors that can report on memory usage, but it would be helpful to have an idea in advance in order to choose an appropriate number of nodes/cores to run a problem.

**Comment by GiudGiud** [comment, marked as answer]
Hello

 I would say at the very least `> 3*(n_dofs_primary + n_dofs_aux) *sizeof(double)`

But it'll depend on which numerical methods you are using to solve the system. A `LU` preconditioner would mean a full matrix to store, some preconditioners will include subsolves which will also require memory, a Krylov method would mean storing several basis vectors to represent the solution.
It'll depend if you use stateful properties or not, and for sure other things. 

It should be fairly easy to determine this experimentally for your case by using uniform refinement and some of the memory postprocessors
https://mooseframework.inl.gov/source/postprocessors/MemoryUsage.html

@YaqiWang may have developed a measure for his application

Guillaume

**Reply by hsheldon** [reply]
Thanks, that's helpful. It hadn't occurred to me that the aux vars were taking up so much memory.
